File: Solutions/Fizz_Buzz/Fizz_Buzz.js
var fizzBuzz = function(n) {
    return _.times(n, x => {
        x = x + 1;
        let str = '';

        if (x % 3 === 0) str += 'Fizz';
        if (x % 5 === 0) str += 'Buzz';

        return str || String(x);
    })
};

var fizzBuzz = function(n) {
    // create a result array of n length
    // loop from 1 to n
        // check if i is divided by both 3 & 5- put "FizzBuzz"
        // else if i is divide by 3 put "Fizz"
        // else if i is divide by 5 put "Buzz"
        // else put i (convert to string) in the array
    
    const res = new Array(n);
    for (let i = 1; i <= n ; i++){
        if(i % 15 === 0) res[i-1] = 'FizzBuzz';
        else if(i % 3 === 0) res[i-1] = 'Fizz';
        else if(i % 5 === 0) res[i-1] = 'Buzz';
        else res[i-1] = i.toString();
    }
    return res;
};

File: Solutions/Fizz_Buzz/Fizz_Buzz.cpp
class Solution {
public:
    vector<string> fizzBuzz(int n) {
        vector<string> result;
        for(int i = 1; i <= n; i++){
            if(i%3 == 0 && i%5 == 0)
                result.push_back("FizzBuzz");
            else if(i%3 == 0)
                result.push_back("Fizz");
             else if(i%5 == 0)
                result.push_back("Buzz");
            else
                result.push_back(to_string(i));
        }
        return result;
    }
}; 

//Fastest
class Solution {
public:
    vector<string> fizzBuzz(int n) {
        bool flag;
        vector<string> output;
        string currOutput;
        for(int i = 1; i<=n; i++) {
            // flag = false;
            currOutput = "";
            
            if(i%3 == 0) {
                currOutput.append("Fizz");
                // flag = true;
            }
            if(i%5 == 0) {
                currOutput.append("Buzz");
                // flag = true;
            }
            if(!currOutput.size()) {
                currOutput.append(to_string(i));
                cout<<to_string(i);
            }
            output.push_back(currOutput);
        }
        return output;
    }
};